generate-renode-scripts.py

```diff
@@ -142,18 +154,36 @@ def generate_silencer(peripheral, shadow_base, **kwargs):
 """.format(peripheral['address'], peripheral['name'])
 
 
-def generate_cpu():
+def generate_cpu(time_provider):
     """ Generates definition of a CPU.
 
     Returns:
         string: repl definition of the CPU
     """
     kind = constants['config_cpu_type']['value']
+    if 'config_cpu_variant' in constants:
+        variant = constants['config_cpu_variant']['value']
+    else:
+        variant = None
 
     if kind == 'VEXRISCV':
-        return """
+        result = """
 cpu: CPU.VexRiscv @ sysbus
 """
+        if variant == 'LINUX':
+            result += """
+    cpuType: "rv32ima"
+    privilegeArchitecture: PrivilegeArchitecture.Priv1_10
+"""
+        else:
+            result += """
+    cpuType: "rv32im"
```

Comment:
I'm not sure when `m` is enabled?

Reply:
According to [this code](https://github.com/enjoy-digital/litex/blob/master/litex/soc/cores/cpu/vexriscv/core.py#L35), it is enabled for variant "standard" and up.

Reply:
So, should probably match that...

I wonder if that information should end up in the headers at all?

Reply:
You can obtain this information by reading `MISA` CSR in RISC-V, so from the software-perspective it does not have to be included in the headers, I think.

Reply:
@mateusz-holenko But that doesn't help compile time optimization?